runners/reference/java/build.gradle

```diff
@@ -27,8 +27,11 @@ framework to execute user-definied functions."""
 dependencies {
   shadow project(path: ":beam-model-pipeline", configuration: "shadow")
   shadow project(path: ":beam-runners-core-construction-java", configuration: "shadow")
+  shadow project(path: ":beam-sdks-java-fn-execution", configuration: "shadow")
   shadow library.java.slf4j_api
-  testCompile library.java.junit
+  shadowTest project(path: ":beam-runners-core-construction-java", configuration: "shadowTest")
```

Comment:
Why is this here? We should not have these kinds of dependencies except for the purpose of running test suites.

Reply:
(is it pulling in NeedsRunner tests? if so, comment that, since these dependencies are by default an incorrect thing to do)

Reply:
No. This is pulling in `InMemoryArtifactStagerService`, which is a test _utility_ rather than a test itself. That class should really live under the main source set in a `testing` subpackage, but I didn't want to do that refactor here.

Is there a tracking bug for removing cross-module test dependencies?

Reply:
Yes - https://issues.apache.org/jira/browse/BEAM-3138

Reply:
For now, IMO it is fine to just move it to `src/main/java` and mark `@Internal` if you don't want to do any other edits.